File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<Integer>();
        int x=0,y=0,val=0;
        for(String token: tokens){
            if(token.equals("+") || token.equals("-") || token.equals("*") || token.equals("/")){
                y = stack.pop();
                x = stack.pop();
                if(token.equals("+")){
                    val = x + y;
                }
                else if(token.equals("-")){
                    val = x - y;
                }
                else if(token.equals("*")){
                    val = x * y;
                }
                else if(token.equals("/")){
                    val = x / y;
                }
                stack.push(val);
            }
            else{
                stack.push(Integer.parseInt(token));
            }
        }
        return stack.peek();
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for i in range(len(tokens)):
            ch = tokens[i]
            if ch == '+' or ch == '-' or ch == '*' or ch == '/':
                a = stack.pop()
                b = stack.pop()
                if ch == '+':
                    stack.append(b + a)
                elif ch == '-':
                    stack.append(b - a)
                elif ch == '*':
                    stack.append(b * a)
                else:
                    stack.append(math.trunc(b / a))
            else:
                val = int(ch)
                stack.append(val)
        return stack.pop()